weaver/wps_restapi/jobs/jobs.py

```diff
@@ -194,44 +203,64 @@ def get_queried_jobs(request):
     """
     Retrieve the list of jobs which can be filtered, sorted, paged and categorized using query parameters.
     """
+
     settings = get_settings(request)
     service, process = validate_service_process(request)
-    detail = asbool(request.params.get("detail", False))
-    page = request.params.get("page", "0")
-    page = int(page) if str.isnumeric(page) else 0
-    limit = request.params.get("limit", "10")
-    limit = int(limit) if str.isnumeric(limit) else 10
-    email = request.params.get("notification_email", None)
-    filters = {
-        "page": page,
-        "limit": limit,
-        # split by comma and filter empty stings
-        "tags": list(filter(lambda s: s, request.params.get("tags", "").split(","))),
-        "access": request.params.get("access", None),
-        "status": request.params.get("status", None),
-        "sort": request.params.get("sort", sort.SORT_CREATED),
-        "notification_email": encrypt_email(email, settings) if email else None,
-        # service and process can be specified by query (short route) or by path (full route)
-        "process": process,
-        "service": service,
-    }
-    groups = request.params.get("groups", "")
-    groups = groups.split(",") if groups else None
-    store = get_db(request).get_store(StoreJobs)
-    items, total = store.find_jobs(request=request, group_by=groups, **filters)
-    body = {"total": total}
 
-    def _job_list(jobs):
-        return [j.json(settings) if detail else j.id for j in jobs]
+    filters = {**request.params, "process": process, "provider": service}
+
+    filters["detail"] = asbool(request.params.get("detail"))
+
+    if request.params.get("datetime", False):
+        # replace white space with '+' since request.params replaces '+' with whitespaces when parsing
+        filters["datetime"] = request.params["datetime"].replace(" ", "+")
+
+    try:
+        filters = sd.GetJobsQueries().deserialize(filters)
+    except Invalid as ex:
+        raise HTTPUnprocessableEntity(json={
+            "code": Invalid.__name__,
+            "description": str(ex)
+        })
 
-    if groups:
-        for grouped_jobs in items:
-            grouped_jobs["jobs"] = _job_list(grouped_jobs["jobs"])
-        body.update({"groups": items})
     else:
-        body.update({"jobs": _job_list(items), "page": page, "limit": limit})
-    body = sd.GetQueriedJobsSchema().deserialize(body)
-    return HTTPOk(json=body)
+
+        detail = filters.pop("detail", False)
+        groups = filters.pop("groups", "").split(",") if filters.get("groups", False) else filters.pop("groups", None)
+
+        filters["tags"] = list(filter(lambda s: s, filters["tags"].split(",") if filters.get("tags", False) else ""))
+        filters["notification_email"] = encrypt_email(
+            filters["notification_email"], settings) if filters.get("notification_email", False) else None
```

Comment:
Sorry I didn't flag this before. 
While you are editing this file, can you adjust this line like follows to help readability: 

```
filters["notification_email"] = (
    encrypt_email(filters["notification_email"], settings) 
    if filters.get("notification_email", False) else None
)            
```